chain-evm/src/state/trie.rs

```diff
@@ -47,6 +47,44 @@ impl<K: Clone + Hash + Eq, V: Clone> Trie<K, V> {
             }
         }
     }
+
+    /// Put a value into a trie, replacing an existing value if there was any.
+    /// This method is useful when you have a mutable reference to the Trie.
+    ///
+    /// This method is similar to `Trie::put`, except that it uses `&mut self` to
+    /// modify the inner state.
+    pub fn insert_or_update(&mut self, key: K, value: V) {
```

Comment:
As `Trie` is a thin wrapper over `Hamt`, I think it makes sense to preserve the ownership pattern: the new state should be returned rather than assigned in place.

Reply:
Should be resolved in https://github.com/input-output-hk/chain-libs/pull/609/commits/2e2c5103f15d6250238c99a5a9f3fdccfc0d7deb